cmd/crank/install.go

```diff
@@ -141,8 +145,19 @@ func (c *installProviderCmd) Run(k *kong.Context) error {
 	kube := typedclient.NewForConfigOrDie(ctrl.GetConfigOrDie())
 	res, err := kube.Providers().Create(context.Background(), cr, metav1.CreateOptions{})
 	if err != nil {
+		warnOutdated(err)
 		return errors.Wrap(err, "cannot create provider")
 	}
 	_, err = fmt.Fprintf(k.Stdout, "%s/%s created\n", strings.ToLower(v1.ProviderGroupKind), res.GetName())
 	return err
 }
+
+func warnOutdated(err error) {
+	serr, ok := err.(*apierrors.StatusError)
+	if !ok {
+		return
+	}
+	if serr.ErrStatus.Code == http.StatusNotFound {
+		fmt.Printf("kubectl crossplane: warning: kubectl-crossplane plugin might be out of date (installed version: %s)\n", version.New().GetVersionString())
```

Comment:
@khos2ow can we send this to `k.Stdout`?

Reply:
I cannot think of a reason why we couldn't. I had thought about it but wanted to get the general feel of this approach before trying that.

Reply:
Oh yes I know we can :) It is nice to do so because you can plumb down a different stdout for testing :+1: I think the solution you have here in general sounds good to me for now. I still feel the implementation of the various CLI methods is a little sloppy (my fault), but it works as expected

Reply:
Refactored (might be a bit too much!)